Gambot.Modules.Roll/DiceEvaluator.cs

```diff
@@ -0,0 +1,271 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Gambot.Modules.Roll
+{
+    public class DiceToken
+    {
+        public bool IsNumber { get; set; }
+        public string StringValue { get; set; }
+        public int IntValue { get; set; }
+        public DiceToken PrecedingToken { get; set; }
+        public DiceToken ProceedingToken { get; set; }
+    }
+
+    public class DiceExpressionResults
+    {
+        public bool IsValid { get; set; } = false;
+        public float Result { get; set; } = 0;
+        public List<string> RollDescriptionList { get; set; } = new List<string>();
+        public string IndividualRollResultString { get; set; } = null;
+    }
+
+    /*
+     * Credits to Simon Gellis for writing the typescript version
+     * https://gist.github.com/SupernaviX/2a4ee385026ccc6566203db6c7464f01
+     */
+    public class DiceEvaluator
+    {
+        private static Random _RNGesus = new Random();
+
+        private DiceExpressionResults _results;
+
+        public DiceEvaluator()
+        {
+            _results = new DiceExpressionResults();
+        }
+
+        private int RollDie(int numberOfSides)
+        {
+            return _RNGesus.Next(numberOfSides) + 1;
+        }
+
+        private IEnumerable<int> RollDice(int numberOfDice, int numberOfSides)
+        {
+            List<int> rolls = new List<int>();
+            for(var i = 0; i < numberOfDice; ++i)
```

Comment:
Not a big deal, but I'd prefer a space after `for` and the body wrapped in `{}`. I believe the repo has ReSharper formatting rules to that effect. If you have access to ReSharper you should try applying them, but it's not a big deal if you don't